src/dune/file_tree.ml

```diff
@@ -193,6 +193,14 @@ let readdir path =
     }
     |> Result.ok
 
+let get_vcs ~default:vcs ~path ~dirs =
+  let dirs =
+    List.map dirs ~f:(fun (name, _, _) -> name) |> String.Set.of_list
```

Comment:
The old code was more efficient. Here we do a map followed by a set conversion just to do two `String.Set.mem` after. To avoid duplicating the string matching, what about adding a function `Vcs.Kind.of_filename`?